app/src/main/java/com/example/scoutconcordia/ShuttleInfo.java

```diff
@@ -0,0 +1,542 @@
+
+                break;
+
+            case "Wednesday":
+                shuttleTimes = getSGWMondayToThursdayTimes();
+
+                for (int i = 0; i < shuttleTimes.length; i++){
+                    if (timeOfDay < shuttleTimes[i]){
+                        nextShuttleTime = shuttleTimes[i];
+                        indexPosition = i;
+                        break;
+                    }
+                }
+
+                messageToUser = "The current time is: " + timeString + " (" + timeOfDay + ")" + " and the next bus shuttle comes at: " + schedule_MonToThurs_SGW[indexPosition] + " (" + nextShuttleTime + ")";
+
+                break;
+
+            case "Thursday":
+                shuttleTimes = getSGWMondayToThursdayTimes();
+
+                for (int i = 0; i < shuttleTimes.length; i++){
+                    if (timeOfDay < shuttleTimes[i]){
+                        nextShuttleTime = shuttleTimes[i];
+                        indexPosition = i;
+                        break;
+                    }
+                }
+
+                messageToUser = "The current time is: " + timeString + " (" + timeOfDay + ")" + " and the next bus shuttle comes at: " + schedule_MonToThurs_SGW[indexPosition] + " (" + nextShuttleTime + ")";
+
+                break;
+
+            case "Friday":
+
+                shuttleTimes = getSGWFridayTimes();
+
+                for (int i = 0; i < shuttleTimes.length; i++){
+                    if (timeOfDay < shuttleTimes[i]){
+                        nextShuttleTime = shuttleTimes[i];
+                        indexPosition = i;
+                        break;
+                    }
+                }
+
+                messageToUser = "Today is " + today + ". The current time is: " + timeString + " (" + timeOfDay + ")" + " and the next bus shuttle comes at: " + schedule_friday_SGW[indexPosition] + " (" + nextShuttleTime + ")";
+
+                break;
+
+
+            case "Saturday":
+                messageToUser = "There are no bus shuttles on Saturdays!";
+                break;
+        }
+
+        // The shuttle time is hardcoded at 30 minutes, because of fluctuations, it is an estimation
+        double estimatedTime =  Math.round((((nextShuttleTime - timeOfDay)*60 ) + 30) * 100.0) / 100.0;
+
+        String overallEstimation = "The estimated route time is: " + estimatedTime;
+
+        System.out.println((messageToUser + "\n" + overallEstimation + " minutes"));
+
+//            return (messageToUser + "\n" + overallEstimation);
+
+        return estimatedTime;
+
+    }
+
+
+
+
+    // This method checks the time and day and finds the next relevant shuttle bus time, from Loyola.
+    // From there, it builds an estimate of how long it would take for the entire trip.
+
+    public double getNextEarliestTimeFromLoyola() {
```

Comment:
In the attributes, times are stored as Strings. However, all the functions return a double or double[] array.

Can you explain why you made this decision and what the difference between the two is?

What would a returned double value look like? What about a double[] array?

Reply:
> In the attributes, times are stored as Strings. However, all the functions return a double or double[] array.
> 
> Can you explain why you made this decision and what the difference between the two is?
> 
> What would a returned double value look like? What about a double[] array?

The reason behind it is that, while a time like "16:00" makes sense to a reader, it is harder to do math logic with that, so there is some code here which coverts the time String into a corresponding double.

What this means is, the double will hold the minutes as a decimal. We need to do a conversion here because minutes are on 60, however decimals are on 100. So while a user understands that 16:05 is before 17:00, we cannot communicate this mathematically. In the Java environment, we would rather say that "16.08 < 17.00".

At the same time, a user probably doesn't want the time in decimals, hence why we keep the strings of the time format in case we want to display it to the user.

Reply:
If I understand correctly, the double values can range from 0.00 to 23.59, correct? Is midnight 0.00 or 24.00 in your implementation?

You may have considered manipulating time-related functions, but if the math checks out then it should be fine. I assume that adding 30 minutes to "13.45" would result in "14.15" -- is this indeed the case?

Reply:
> If I understand correctly, the double values can range from 0.00 to 23.59, correct? Is midnight 0.00 or 24.00 in your implementation?
> 
> You may have considered manipulating time-related functions, but if the math checks out then it should be fine. I assume that adding 30 minutes to "13.45" would result in "14.15" -- is this indeed the case?

That should indeed be the case, yes